specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2021-01-25/guestconfiguration.json

```diff
@@ -0,0 +1,1456 @@
+          "enum": [
+            "Consistency",
+            "Initial"
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssignmentReportResource"
+          },
+          "description": "The list of resources for which guest configuration assignment compliance is checked."
+        }
+      }
+    },
+    "GuestConfigurationAssignmentProperties": {
+      "properties": {
+        "targetResourceId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "VM resource Id.",
+          "x-nullable": true
+        },
+        "guestConfiguration": {
+          "$ref": "#/definitions/GuestConfigurationNavigation",
+          "description": "The guest configuration to assign."
+        },
+        "complianceStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A value indicating compliance status of the machine for the assigned guest configuration.",
+          "x-ms-enum": {
+            "name": "complianceStatus",
+            "modelAsString": true
+          },
+          "enum": [
+            "Compliant",
+            "NonCompliant",
+            "Pending"
+          ]
+        },
+        "lastComplianceStatusChecked": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Date and time when last compliance status was checked.",
+          "x-nullable": true
+        },
+        "latestReportId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Id of the latest report for the guest configuration assignment. ",
+          "x-nullable": true
+        },
+        "latestAssignmentReport": {
+          "$ref": "#/definitions/AssignmentReport",
+          "description": "Last reported guest configuration assignment report."
+        },
+        "context": {
+          "type": "string",
+          "description": "The source which initiated the guest configuration assignment. Ex: Azure Policy"
+        },
+        "assignmentHash": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Combined hash of the configuration package and parameters.",
+          "x-nullable": true
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state, which only appears in the response.",
+          "x-nullable": true,
+          "x-ms-enum": {
+            "name": "provisioningState",
+            "modelAsString": true
+          },
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Created"
+          ]
+        },
+        "resourceType": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of the resource - VMSS / VM",
+          "x-nullable": true
+        },
+        "vmssVMList": {
```

Comment:
Can you clarify why the guestconfigassignment needs all of the info for each VM instance within the VMSS instead of querying off of each instance itself?

Reply:
Guest Configuration will be auditing different settings on the VM for VMSS.
So when we get assignment information for VMSS - we want to provide the compliance status of the VMSS as well as details of the VM compliance status.
VM instance information include basic information like VMUID, Compliance Status, Latest Report ID...